web-bluetooth/read-characteristic-value-changed.js

```diff
@@ -1,13 +1,34 @@
+var bluetoothDevice;
 var batteryLevelCharacteristic;
 
 function onReadBatteryLevelButtonClick() {
-  log('Requesting Bluetooth Device...');
-  navigator.bluetooth.requestDevice(
-    {filters: anyDevice(), optionalServices: ['battery_service']})
-  .then(device => {
-    log('Connecting to GATT Server...');
-    return device.gatt.connect();
+  let requestDevice = Promise.resolve();
```

Comment:
Consider extracting the requestDevice() stuff into a helper function too. It'll both simplify reading this function, and help in cases where multiple buttons need to be able to request the device.